test/lib/jdk/test/lib/SA/SATestUtils.java

```diff
@@ -77,11 +78,54 @@ public static void skipIfCannotAttach() {
 
     /**
      * Returns true if this platform is expected to require extra privileges (running using sudo).
+     * If we are running as root or developer mode is enabled, then sudo is not needed.
      */
     public static boolean needsPrivileges() {
-        return Platform.isOSX() && !Platform.isRoot();
+        if (!Platform.isOSX()) {
+            return false;
+        }
+        if (Platform.isRoot()) {
+            return false;
+        }
+        return !developerModeEnabled();
     }
 
+    /*
+     * Run "DevToolsSecurity --status" to see if developer mode is enabled.
+     */
+    private static boolean developerModeEnabled() {
+        List<String> cmd = new ArrayList<String>();
+        cmd.add("DevToolsSecurity");
```

Comment:
Is this tool likely to be in the path?

Reply:
It is for my work machine, my home machine, and the CI machines. If not installed then an exception is thrown. It probably requires that xcode be installed, but this is already a requirement to run the tests.